src/main/java/org/opensearch/flowframework/workflow/ToolStep.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.flowframework.workflow;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.flowframework.exception.FlowFrameworkException;
+import org.opensearch.ml.common.agent.MLToolSpec;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.CompletableFuture;
+
+import static org.opensearch.flowframework.common.CommonValue.DESCRIPTION_FIELD;
+import static org.opensearch.flowframework.common.CommonValue.INCLUDE_OUTPUT_IN_AGENT_RESPONSE;
+import static org.opensearch.flowframework.common.CommonValue.NAME_FIELD;
+import static org.opensearch.flowframework.common.CommonValue.PARAMETERS_FIELD;
+import static org.opensearch.flowframework.common.CommonValue.TOOLS_FIELD;
+import static org.opensearch.flowframework.common.CommonValue.TYPE;
+import static org.opensearch.flowframework.util.ParseUtils.getStringToStringMap;
+
+/**
+ * Step to register a tool for an agent
+ */
+public class ToolStep implements WorkflowStep {
+
+    private static final Logger logger = LogManager.getLogger(ToolStep.class);
+    CompletableFuture<WorkflowData> toolFuture = new CompletableFuture<>();
+    static final String NAME = "tool";
```

Comment:
nit: should this be changed to `register_tool` or `create_tool`? we have that pattern for most other steps

Reply:
Changed to `create_tool`